### PR TITLE
Recipes

### PR DESCRIPTION
**Pull request by stefanwalther**
Hi,

here's a PR for two recipes:
- default-layout 
- Assemble and browser-sync (to serve files and watch/update content & css files)

In general I found it extremely useful if one is able to execute the recipes immediately (with small, fully working examples), so I have created this page to test the recipes: https://github.com/stefanwalther/assemble-recipes

Regards
Stefan

### CODE REVIEW

**Comment by jonschlinkert** [comment]
these are great thanks! would you mind updating the javascript code style to be consistent with the code in the library? maybe just remove the extra padding around objects etc. Or I can do it if you want.

**Comment by jonschlinkert** [comment]
the assemble-recipes repo is great too, I'm planning on moving the recipes/docs/examples from assemble-core to this repo, any chance you would be interested in moving those recipes here? No worries if you'd prefer to maintain them separately, that's just as great :)

**Comment by stefanwalther** [comment]
Sure, can do, didn't realized that the formatting is different, will do so.

**Comment by stefanwalther** [comment]
I am already moving some of the recipes there as you have seen, basically it's great for me to learn assemble/assemble-core, etc.
... and what I like is that there is test-coverage, so also in future versions of assemble all these small projects can be tested, so one (or you) can ensure that all recipes still work.

**Comment by stefanwalther** [comment]
What about breaking out the recipes completely from assemble/assemble ? Then you could have the test-coverage there + documentation. You could then fetch the contents using e.g. github-content or stefanwalther/sammler to include these files into the documentation (that's how I am planning to do with a lot of repos: fetching content from several repos into a single web-site).

**Comment by jonschlinkert** [comment]
> fetching content from several repos into a single web-site

sure, that's been the plan all along (for all of our projects). It's still the goal, but it's worth noting that it's also the reason why it seems like we don't have any docs. All of the libs that are used in assemble, verb and elsewhere typically have complete docs, but pulling them together will unfortunately be the very last thing that happens - so it seems like they don't exist. I'm okay with that if you are! :)

would you like to move assemble-recipes to the assemble org and maintain it? again please don't feel awkward turning that down. I'm going to add you to the org anyway, and it's totally cool either way.

**Comment by stefanwalther** [comment]
> but it's worth noting that it's also the reason why it seems like we don't have any docs

I have exactly the same problem ;-) Main reason to wait for assembe and not using other static-site-generators ;-)

> would you like to move assemble-recipes to the assemble org and maintain it?
Sure, np

**Comment by stefanwalther** [comment]
Doesn't work:

image

image

**Comment by jonschlinkert** [comment]
you should, I'm guessing you accepted the invitation already? I'll go look at permission settings too

**Comment by jonschlinkert** [comment]
try again

**Comment by stefanwalther** [comment]
:+1: , worked now, transferred.

**Comment by jonschlinkert** [comment]
awesome! thanks!

File: docs/recipes/browser-sync.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	color: @mainColor;
+}
+
+h1 {
+	color: @secondColor;
+}
+```
+
+**less/variables.less**
+```css
```
Comment (jonschlinkert) [inline docs/recipes/browser-sync.md]: one other small nitpick, would you mind adding an extra line between headings/paragraphs and code blocks? fwiw I prefer the way you did it personally, but the syntax highlighting in some ST themes doesn't work unless there is an extra line. (lame, but true)

Comment (stefanwalther) [inline docs/recipes/browser-sync.md]: Sure, will do.

Comment (jonschlinkert) [inline docs/recipes/browser-sync.md]: here, to show you what I mean:

screen shot 2016-01-04 at 10 31 02 am

versus 

screen shot 2016-01-04 at 10 31 41 am

Comment (jonschlinkert) [inline docs/recipes/browser-sync.md]: thx

**Comment by stefanwalther** [comment]
So, now I am confused. 
Do want to add all recipes to assemble-recipes (also those from assemble-core) or keep them as duplicates, so merging this PR and keeping the same recipes with tests also in assemble-recipes?

**Comment by jonschlinkert** [comment]
> Do want to add all recipes to assemble-recipes (also those from assemble-core) or keep them as duplicates, so merging this PR and keeping the same recipes with tests also in assemble-recipes?

oh yeah I see what you mean. want to just close this and focus efforts over on assemble-recipes?

**Comment by jonschlinkert** [comment]
sorry i'm getting ready to walk out the door so I'm a little distracted. also I have a bunch of documentation changes already done that I'll be pushing up later. afterwards we can move any of that over to assemble-recipes or whatever makes sense.

**Comment by jonschlinkert** [comment]
wait I thought we weren't going to merge this?

**Comment by stefanwalther** [comment]
;-((( Total confusion now ... tell me what to do, can remove it, too.
I on my part will continue to work on the recipes-repo ... Sorry ... ;-)

**Comment by jonschlinkert** [comment]
It's fine, no reason to be sorry. I think it was probably just a misunderstanding (that isn't of any consequence, so don't lose sleep over it!)

just for the sake of clarifying, you asked:

> Do want to add all recipes to assemble-recipes (also those from assemble-core) or keep them as duplicates...

Then I replied:

> oh yeah I see what you mean. want to just close this and focus efforts over on assemble-recipes?

Then you merged. But I didn't care either way, I just thought _you were saying it didn't make sense_

**Comment by stefanwalther** [comment]
Yes, you are right or not ... ;-)
Just wanted to close this to continue working on the recipes, actually **moving** stuff to the recipes now ... which will mean that the more accurate and more complete examples are in recipes.

**Is that OK for you now?**

Then the recipes could be easily included into the documentation. Benefit of the extra repo approach with recipes is, that you have encapsulated fully working integration tests there ... + the documentation of the recipe.

**Comment by jonschlinkert** [comment]
> Is that OK for you now?

Can you wait a little bit? I have a bunch of stuff to push up and would like to avoid merge conflicts. might be a couple hours.

**Comment by stefanwalther** [comment]
Sure, np.

**Pull request merged**